python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt/sling/resources.py

```diff
@@ -356,23 +362,50 @@ def sync(
             yield from self._exec_sling_cmd(cmd, encoding=encoding)
 
     @public
+    @deprecated_param(
+        param="dagster_sling_translator",
+        breaking_version="2.0",
+        additional_warn_text="Param is only required in `sling_assets` decorator.",
+    )
+    @deprecated_param(
+        param="replication_config",
+        breaking_version="2.0",
+        additional_warn_text="Param is only required in `sling_assets` decorator.",
+    )
```

Comment:
I don't think these should be deprecated. These arguments should not be needed in an asset context, but if anyone wants to run this in an op, they will need to pass these parameters manually, as they will not be available in the op context.

We should add a test where this resource is executed in an `@op` to ensure that we support this behavior.

Reply:
That's a good point. I'm not sure if the Sling implementation currently supports used within an `@op`. I just wrote a unit test for it, and get the following error. I'll look into it some more.
```
self = <dagster._core.execution.context.compute.OpExecutionContext object at 0x115e99b10>

    @public
    @property
    def assets_def(self) -> AssetsDefinition:
        """The backing AssetsDefinition for what is currently executing, errors if not available."""
        assets_def = self.job_def.asset_layer.assets_def_for_node(self.node_handle)
        if assets_def is None:
>           raise DagsterInvalidPropertyError(
                f"Op '{self.op.name}' does not have an assets definition."
            )
E           dagster._core.errors.DagsterInvalidPropertyError: Op 'my_sling_op_yield_events' does not have an assets definition.

../../../.direnv/python-3.11/lib/python3.11/site-packages/dagster/_core/execution/context/compute.py:580: DagsterInvalidPropertyError
```
@PedramNavid, let me know if I'm missing something silly.

Reply:
Doh, it was how we're getting the metadata, please disregard. I'll apply a fix.
```
        # retrieve decorator params from context
        metadata_by_key = context.assets_def.metadata_by_key
        first_asset_metadata = next(iter(metadata_by_key.values()))
```

Reply:
@rexledesma thanks again. https://github.com/dagster-io/dagster/pull/20564/commits/a286b9f8884910a34f50d0c92ac3ddc2f2dde26c adds unit test for op usage, and removes deprecation warnings.